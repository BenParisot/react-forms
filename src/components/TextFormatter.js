import React, { PureComponent } from 'react';

export default class TextFormatter extends PureComponent {
    state = {
      text: '',
      color: 'black'
    }
    handleTextChange = ({ target }) => {
      this.setState({ text: target.value });
    }

    handleColorChange = ({ target }) => {
      this.setState({ color: target.value });
    }

    render() {
      const {
        text
      } = this.state;
      return (
          <>
            <form>
              <input onChange={this.handleTextChange}></input>
              <input type="color" onChange={this.handleColorChange}></input>
            </form>
            <p style={{ color:this.state.color }}>{text}</p>  
        </> 
      );
    }
}
