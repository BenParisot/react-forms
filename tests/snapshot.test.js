import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from '../src/components/TextFormatter';

describe('TextFormatter component', () => {
  it('formats text from user input', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
