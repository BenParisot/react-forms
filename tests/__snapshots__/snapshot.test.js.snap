// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatter component formats text from user input 1`] = `
<Fragment>
  <form>
    <input
      onChange={[Function]}
    />
    <input
      onChange={[Function]}
      type="color"
    />
  </form>
  <p
    style={
      Object {
        "color": "black",
      }
    }
  />
</Fragment>
`;
